### PR TITLE
Add a feature to write separate routes in multiple files 

### PR DESCRIPTION
**Pull request by tulayang**
I revised some contents. Now, we can write separate routes in multiple ``routes`` or multiple files. 

For example, this is the directory structure of our project:

```
app
|- routes1.nim
|- routes2.nim
|- server.nim
```

routes1.nim:

```nim
routes:
  get "/route1":
    resp "get /route1"
  post "/route1":
    resp "post /route1"
```

routes2.nim:

```nim
routes:
  get "/route2":
    resp "get /route2"
  post "/route2":
    resp "post /route2"
```

server.nim:

```nim
import jester, asyncdispatch

settings:
  bindAddr = "127.0.0.1"
  port = Port(9001)

include "routes1.nim"
include "routes2.nim"

waitFor jester.serve(settings)
```

### CODE REVIEW

**Comment by tulayang** [comment]
The complied routes look like this:

```nim
block:
    proc match(request: Request; response: jester.Response): Future[bool] {.async.} =
      setDefaultResp()
      case request.reqMeth
      of HttpGet:
        let patternMatchRet = (re.match|re.match|patterns.match)(
            @ [Node(typ: NodeText, text: "/route2", optional: false)],
            request.pathInfo)
        block thisRoute:
            if patternMatchRet.matched:
              copyParams(request, patternMatchRet.params)
              resp "get /route2"
              if checkAction(response):
                return true
      of HttpPost:
        let patternMatchRet = (re.match|re.match|patterns.match)(
            @ [Node(typ: NodeText, text: "/route2", optional: false)],
            request.pathInfo)
        block thisRoute:
            if patternMatchRet.matched:
              copyParams(request, patternMatchRet.params)
              resp "post /route2"
              if checkAction(response):
                return true
      of HttpPut:
      of HttpDelete:
      of HttpHead:
      of HttpOptions:
      of HttpTrace:
    
    jester.addMatch(match)
```

**Comment by dom96** [comment]
Thank you for this. I can't review it right now so please give me a reminder if I forget to do it in the next couple of weeks!

**Comment by tulayang** [comment]
ok

File: jester.nim [hunk 1]
```diff
@@ -82,6 +82,12 @@ type
 
 const jesterVer = "0.1.0"
 
+var jes: Jester
```
Comment (Xe) [inline jester.nim]: I understand why you are doing this, but could there be another way to do this without module-level mutable state?

**Comment by tulayang** [comment]
@Xe 

Did you worry about the side effects of the global variable ? There is no need to worry about this. Outside jester, it can't be accessed.

In addition, this can simplify the user's creation process. They don't need to care about how to initialize.

**Comment by yglukhov** [comment]
@tulayang, is there a way to avoid includes? Can we use import instead?

**Comment by tulayang** [comment]
@yglukhov

Yes，it's ok.

routes1.nim:

```nim
import jester, asyncdispatch

routes:
  get "/route1":
    resp "get /route1"
  post "/route1":
    resp "post /route1"
```

routes2.nim:

```nim
import jester, asyncdispatch

routes:
  get "/route2":
    resp "get /route2"
  post "/route2":
    resp "post /route2"
```

server.nim:

```nim
import jester, asyncdispatch
import routes1, routes2

settings:
  bindAddr = "127.0.0.1"
  port = Port(9001)

waitFor jester.serve(settings)
```

**Comment by yglukhov** [comment]
Thats so cool!

**Comment by dom96** [comment]
Very nice. But I am also concerned about the global `jester` variable. Is there nothing we could do to prevent it?

**Comment by tulayang** [comment]
@dom96 

In order to separate routes, a queue is required to hold each ``match``.  The only possible way to avoid ``jester`` variable is  that let the user to initialize ``jester`` variable.

```nim
var jes = newJester()

routes: ...

routes: ...

waitFor jester.serve()
```

**Comment by iwanbk** [comment]
@tulayang i wonder why you close this PR? It is very useful and cool PR.

**Pull request closed without merging**